codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -170,6 +170,7 @@ public ChannelFuture writeHeaders(final ChannelHandlerContext ctx, final int str
                                 "Stream %d in unexpected state: %s", stream.id(), stream.state()));
                 }
             }
+            stream.headersSent();
```

Comment:
If I understand correctly, the case where we pass the headers to the flow controller, they will not get written immediately but enqueued. In this case it seems like we want to ensure that even though we have not technically _sent_ the headers, we still want to make sure that we do not enqueue headers containing a status code as well as they would be transmitted in order and the second header would be in violation of the spec.

Therefore it seems safest (and simplest) to mark headersSent as soon as we had declared out intention to send them. Please let me know if I am missing something here.